### PR TITLE
[C] Handle failed log buffer delete in C Media Driver

### LINKED ISSUE #620: Handle failed log buffer delete in C Media Driver
The Java Media Driver attempts the delete the log buffer during the slow tick of the conductor duty cycle and if it fails it increments the FREE_FAILS counter and tries again in the next cycle. The C Media Driver should do similar to support Windows.

### PR DESCRIPTION
**Pull request by vyazelenko**
Fixes #620.

### CODE REVIEW

File: aeron-client/src/main/c/util/aeron_fileutil.c [hunk 1]
```diff
@@ -530,25 +530,33 @@ int aeron_map_existing_log(aeron_mapped_raw_log_t *mapped_raw_log, const char *p
 
 int aeron_map_raw_log_close(aeron_mapped_raw_log_t *mapped_raw_log, const char *filename)
 {
-    int result = 0;
-
-    if (mapped_raw_log->mapped_file.addr != NULL)
+    if (!aeron_map_raw_log_free(mapped_raw_log, filename))
```
Comment (mjpt777) [inline aeron-client/src/main/c/util/aeron_fileutil.c:533]: Should be `aeron_mapped_raw_log_free`.

File: aeron-client/src/test/c/util/aeron_fileutil_test.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <exception>
+#include <functional>
+
+#include <gtest/gtest.h>
+
+#include "../aeron_client_test_utils.h"
```
Comment (mjpt777) [inline aeron-client/src/test/c/util/aeron_fileutil_test.cpp:22]: Use path from base include in project.

File: aeron-client/src/main/c/util/aeron_fileutil.c (hunk 1 above)
Comment (mjpt777) [reply aeron-client/src/main/c/util/aeron_fileutil.c:533]: I'm doing some renaming and it should be `aeron_raw_log_free`.

Comment (vyazelenko) [reply aeron-client/src/main/c/util/aeron_fileutil.c:533]: @mjpt777 Noticed that and already my branch.

File: aeron-system-tests/src/test/java/io/aeron/archive/ReplicateRecordingTest.java [hunk 3]
```diff
@@ -93,6 +95,7 @@ public void before()
                 .threadingMode(ThreadingMode.SHARED)
                 .errorHandler(Tests::onError)
                 .spiesSimulateConnection(true)
+                .timerIntervalNs(TIMER_INTERVAL_NS)
```
Comment (mjpt777) [inline aeron-system-tests/src/test/java/io/aeron/archive/ReplicateRecordingTest.java:98]: Is this still needed?

Comment (vyazelenko) [reply aeron-system-tests/src/test/java/io/aeron/archive/ReplicateRecordingTest.java:98]: @mjpt777 This was exposing the bug. Without it tests were passing ok.

**Pull request merged**